staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -630,6 +630,36 @@ type PersistentVolumeClaimSpec struct {
 	StorageClassName *string `json:"storageClassName,omitempty" protobuf:"bytes,5,opt,name=storageClassName"`
 }
 
+// PvcConditionType is a valid value of PvcCondition.Type
+type PvcConditionType string
+
+const (
+	// Ready means any pending operation on pvc is successfully completed
+	PvcReady PvcConditionType = "Ready"
+	// An user trigger resize of pvc has been started
+	PvcResizeStarted PvcConditionType = "ResizeStarted"
+	// PvcResizeFailed means user trigged resize of PVC has failed
+	PvcResizeFailed PvcConditionType = "ResizeFailed"
+)
+
+// PvcCondition contails details about state of pvc
+type PvcCondition struct {
+	Type   PvcConditionType `json:"type" protobuf:"bytes,1,opt,name=type,casttype=PvcConditionType"`
+	Status ConditionStatus  `json:"status" protobuf:"bytes,2,opt,name=status,casttype=ConditionStatus"`
+	// Last time we probed the condition.
+	// +optional
+	LastProbeTime metav1.Time `json:"lastProbeTime,omitempty" protobuf:"bytes,3,opt,name=lastProbeTime"`
+	// Last time the condition transitioned from one status to another.
+	// +optional
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty" protobuf:"bytes,4,opt,name=lastTransitionTime"`
+	// Unique, one-word, CamelCase reason for the condition's last transition.
```

Comment:
The description of this should look more like the Event Reason godoc:

```
	// Optional; this should be a short, machine understandable string that gives the reason
	// for this event being generated. For example, if the event is reporting that a container
	// can't start, the Reason might be "ImageNotFound".
```

Saying that it is CamelCase doesn't help a human understand what it is for.  You should describe in this description which conditions are normally set and also explain that other types are possible.

Reply:
added more info there.